### PR TITLE
[264] show error if download failed

### LINKED ISSUE #264: inform the user when the download of a runtime fails
steps:
1. EXEC: pick "Download Runtime" in the context menu of the "Servers" view
2. EXEC: pick a runtime (ex. wildfly-11) & provide answers to all questions (ex. accept the license etc.)
3. ASSERT: you get an info dialog that the runtime is being downloaded
image

4. EXEC: cut the network

Result:
Nothing happens, it seems to silently fail.

Expected result:
We should inform the user that the download failed.

### PR DESCRIPTION
**Pull request by adietish**
fixes #264

### CODE REVIEW

File: src/jobprogress.ts [hunk 1]
```diff
@@ -69,10 +69,29 @@ export class JobProgress {
...
   private onJobRemoved(jobRemoved: Protocol.JobRemoved) {
       if (!this.isJob(jobRemoved.handle)) {
-        return;
+          return;
       }
       this.clearTimeout();
-      this.resolve(this.job);
+      if (!StatusSeverity.isOk(jobRemoved.status)) {
+          vscode.window.showErrorMessage(this.getErrorMessage(jobRemoved.status));
+          this.resolve();
```
Comment (jrichter1) [inline src/jobprogress.ts]: You already have a clause like this for error handling, so why not just reject with the error message?

Comment (adietish) [reply src/jobprogress.ts]: Good catch. Correcting.

Comment (jrichter1) [reply src/jobprogress.ts]: And now you need to adjust the tests :)

Comment (adietish) [reply src/jobprogress.ts]: @jrichter1 they are, build is green :)

File: package.json [hunk 2]
```diff
@@ -235,6 +235,7 @@
     "vscode": "^1.1.21"
   },
   "dependencies": {
+    "chai-as-promised": "^7.1.1",
```
Comment (jrichter1) [inline package.json]: this should be in devDependencies

Comment (adietish) [reply package.json]: corrected, thx for spotting.

**Comment by adietish** [comment]
@jrichter1 please re-review

**Pull request merged**